src/util/error.c

```diff
@@ -31,12 +31,11 @@ int error_set(error *e, int code, const char *msg) {
 
 int error_fmt(error *e, int code, const char *fmt, ...) {
   va_list ap;
-
   e->code = code;
+#ifdef DEBUG
   va_start(ap, fmt);
   vsnprintf(e->msg, ERROR_MSGBUF_LEN, fmt, ap);
   va_end(ap);
-#ifdef DEBUG
```

Comment:
Great, thanks! I would also suggest to look at the flip side of this observation - when in DEBUG mode, useful output is being flooded by thousands of messages about wrong dimensions - if those are not real errors, a different check should be used (or a 'silent' parameter added).